sdk-python/list_and_match_messages.py

```diff
@@ -0,0 +1,54 @@
+import argparse
+from datetime import datetime
+from awaazde.constants import CommonConstants, APIConstants
+from awaazde.utils import CSVUtils
+from awaazde import AwaazDeAPI
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description='Check Created Messages')
+    parser.add_argument('username', type=str, help='Username of the tenant')
+    parser.add_argument('password', type=str, help='Password of the tenant')
+    parser.add_argument('organization', type=str, help='Organization of the tenant')
+    parser.add_argument('match_path', type=str, help='Path for the csv with which filtered data is to be matched')
+    parser.add_argument('--params', type=str,
+                        help='Fields on which filters are to be applied in the form: {"send_on__gt":"10-12-2020",'
+                             '"send_on__lt":"10-12-2020","tags":"dummy_tag"}')
+    parser.add_argument('--match_criteria', type=str,
+                        help='Field to use when matching User record with the queried resultant record',
+                        default=CommonConstants.MESSAGE_ID)
+    args = parser.parse_args()
+    return args
+
+
+if __name__ == '__main__':
+    """
+        Step 1 : Parse the arguments
+    """
+    args = parse_arguments()
+    headers, message_data = CSVUtils.read_csv(args.match_path)
+    awaazde_api = AwaazDeAPI(args.organization, args.username, args.password)
+
+    """
+        Step 2 : Send an api request to  Get List of Messages based on filters
+    """
+    params = {}
+    params["tags"] = CommonConstants.PHONE_NUMBER_FIELD
+    params['page'] = APIConstants.DEFAULT_BULK_CREATE_LIMIT
+    params['fields'] = CommonConstants.PHONE_NUMBER_FIELD, CommonConstants.ID_FIELD, CommonConstants.SEND_ON_FIELD
+    responses = awaazde_api.messages.list_depaginated(params)
+    response_dict = {[response[CommonConstants.PHONE_NUMBER_FIELD]] for response in responses}
```

Comment:
I don't understand what this dict comprehension does but the syntax doesn't look right

Reply:
you can also name this var more descriptively than `response_dict` to help reader understand what this is